kv/txn_coord_sender.go

```diff
@@ -396,14 +396,26 @@ func (tc *TxnCoordSender) sendOne(ctx context.Context, call proto.Call) {
 				return
 			}
 			tc.Lock()
-			if txnMeta := tc.txns[id]; id != "" && txnMeta != nil {
+			txnMeta, metaOK := tc.txns[id]
```

Comment:
If we ever insert a nil pointer in the map we have a leak, since nothing will ever clean up that map entry. I prefer to use the `, ok` lookup to test for existence; if we ever get a nil pointer here the panic will alert us to the fact that something is not as we expect.